### PR TITLE
Fix support of `object`, `list: true` in MDX templates

### PR DESCRIPTION
**Pull request by jeffsee55**
When specifying `list: true` for an `object` in an MDX template there was some strange behavior, it was mapping over the template `fields` rather than the values. Tests previously weren't covering this scenario so those have also been updated.

I also noticed that empty paragraph blocks weren't getting stripped out during stringify, it may be desirable for multiple lines to represent empty paragraphs but it's hard to know when it's intentional. I think it's best to remove them for now

### CODE REVIEW

**Comment by jeffsee55** [comment]
Closes https://github.com/tinacms/tinacms/issues/2287

File: examples/basic/.tina/schema.ts [hunk 1]
```diff
@@ -15,10 +15,32 @@ export default defineSchema({
...
+    //       type: 'string',
+    //       label: 'Blog Post Body',
+    //       name: 'body',
+    //       isBody: true,
+    //       ui: {
+    //         component: 'textarea',
+    //       },
+    //     },
+    //   ],
+    // },
```
Comment (Enigmatical) [inline examples/basic/.tina/schema.ts:38]: If this isn't needed anymore, probably couldn't hurt to just remove it.

**Review by Enigmatical — DISMISSED** [review]
Looks good 👍

**Comment by logan-anderson** [comment]
Looks good to me too. Might have to resolve some conflicts to be merged

**Pull request merged**